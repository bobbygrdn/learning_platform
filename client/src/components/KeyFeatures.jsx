import React from 'react'

export default function KeyFeatures() {
    // TODO: Finish key features component
    return (
        <div>

        </div>
    )
}
