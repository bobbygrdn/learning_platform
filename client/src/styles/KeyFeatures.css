.features {
    display: grid;
    grid-template-rows: repeat(3, 1fr);
    grid-template-columns: repeat(2, 1fr);
    grid-gap: 1em;
    margin-bottom: 5em;
    padding: 0 1em;
}

.icon {
    margin-bottom: 1em;
}

@media screen and (max-width: 600px) {
    .icon {
        width: 4em;
        height: 4em;
    }
}

@media screen and (min-width: 1000px) {
    .text {
        font-size: 1.2rem;
        padding: 0 3.5em;
    }
}