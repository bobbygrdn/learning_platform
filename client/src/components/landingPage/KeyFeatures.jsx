import React from 'react';
import Feature from './Feature';
import '../../styles/landingPage/KeyFeatures.css'

export default function KeyFeatures() {
    const features = [
        {
            description: "Custom learning paths for different goals and skill levels",
            Image: `${process.env.PUBLIC_URL + "/resources/learningPath_icon.png"}`,
        },
        {
            description: "Access a library of courses created by expert programmers",
            Image: `${process.env.PUBLIC_URL + "/resources/library_icon.png"}`,
        },
        {
            description: "Engage with a community of fellow learners and experts",
            Image: `${process.env.PUBLIC_URL + "/resources/forum_icon.png"}`,
        },
        {
            description: "Mobile-first design for on-the-go learning",
            Image: `${process.env.PUBLIC_URL + "/resources/mobile_icon.png"}`,
        },
        {
            description: "Earn badges and certificates to demonstrate your knowledge and progress",
            Image: `${process.env.PUBLIC_URL + "/resources/certificate_icon.png"}`,
        },
        {
            description: "Personalized recommendations based on your learning history and preferences",
            Image: `${process.env.PUBLIC_URL + "/resources/recommendations_icon.png"}`,
        }
    ]
    return (
        <div className='features'>
            {features.map((feature, index) => (
                <Feature key={index} feature={feature} />
            ))}
        </div>
    )
}
