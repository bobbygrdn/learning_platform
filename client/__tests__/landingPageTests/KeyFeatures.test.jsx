import React from 'react';
import { render, screen } from '@testing-library/react';
import KeyFeatures from '../../src/components/landingPage/KeyFeatures.jsx';

const features = () => {
    return [
        {
            description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, quae.',
            image: "someurl/picture"
        },
        {
            description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, quae.',
            image: "someurl/picture"
        },
        {
            description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, quae.',
            image: "someurl/picture"
        },
        {
            description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, quae.',
            image: "someurl/picture"
        },
        {
            description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, quae.',
            image: "someurl/picture"
        },
        {
            description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, quae.',
            image: "someurl/picture"
        }
    ]
}

it('renders without crashing', async () => {
    render(<KeyFeatures />);
    expect(await screen.findAllByTestId('features')).toMatchSnapshot();
})

it('should render six key feature elements', () => {
    render(<KeyFeatures features={features()} />);
    expect(screen.getAllByTestId('feature')).toHaveLength(6);
})