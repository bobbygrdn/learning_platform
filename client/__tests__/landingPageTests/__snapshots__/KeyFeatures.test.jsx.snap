// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Array [
  <div
    class="features"
    data-testid="features"
  >
    <div
      class="feature"
      data-testid="feature"
    >
      <img
        alt="Feature undefined"
        class="icon"
        src="undefined/resources/learningPath_icon.png"
      />
      <p
        class="text"
        data-testid="text"
      >
        Custom learning paths for different goals and skill levels
      </p>
    </div>
    <div
      class="feature"
      data-testid="feature"
    >
      <img
        alt="Feature undefined"
        class="icon"
        src="undefined/resources/library_icon.png"
      />
      <p
        class="text"
        data-testid="text"
      >
        Access a library of courses created by expert programmers
      </p>
    </div>
    <div
      class="feature"
      data-testid="feature"
    >
      <img
        alt="Feature undefined"
        class="icon"
        src="undefined/resources/forum_icon.png"
      />
      <p
        class="text"
        data-testid="text"
      >
        Engage with a community of fellow learners and experts
      </p>
    </div>
    <div
      class="feature"
      data-testid="feature"
    >
      <img
        alt="Feature undefined"
        class="icon"
        src="undefined/resources/mobile_icon.png"
      />
      <p
        class="text"
        data-testid="text"
      >
        Mobile-first design for on-the-go learning
      </p>
    </div>
    <div
      class="feature"
      data-testid="feature"
    >
      <img
        alt="Feature undefined"
        class="icon"
        src="undefined/resources/certificate_icon.png"
      />
      <p
        class="text"
        data-testid="text"
      >
        Earn badges and certificates to demonstrate your knowledge and progress
      </p>
    </div>
    <div
      class="feature"
      data-testid="feature"
    >
      <img
        alt="Feature undefined"
        class="icon"
        src="undefined/resources/recommendations_icon.png"
      />
      <p
        class="text"
        data-testid="text"
      >
        Personalized recommendations based on your learning history and preferences
      </p>
    </div>
  </div>,
]
`;
